### PR TITLE
fix(IAM Policy Management): S2S support for enterprise accounts

### PR DESCRIPTION
**Pull request by Rajesh-Pirati**
## PR summary
Authorization policy type support for enterprise accounts
issue: https://github.ibm.com/IAM/AM-issues/issues/2350

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

## Other information

test results
**example**:
image

**integration**
image

### CODE REVIEW

File: iampolicymanagementv1/iam_policy_management_v1_examples_test.go [hunk 1]
```diff
@@ -742,6 +753,76 @@ var _ = Describe(`IamPolicyManagementV1 Examples Tests`, func() {
 			Expect(response.StatusCode).To(Equal(201))
 			Expect(policyTemplate).ToNot(BeNil())
 		})
+
+		It(`CreatePolicyS2STemplate request example`, func() {
+			fmt.Println("\nCreatePolicyS2STemplate() result:")
+			// begin-create_policy_template
```
Comment (swcolley) [inline iampolicymanagementv1/iam_policy_management_v1_examples_test.go:759]: This has the same name as line 708. Can be an issue with example generation in the docs if this are named same? I think there might be a collison.

File: iampolicymanagementv1/iam_policy_management_v1_examples_test.go [hunk 2]
```diff
@@ -742,6 +753,76 @@ var _ = Describe(`IamPolicyManagementV1 Examples Tests`, func() {
...
+			policyTemplate, response, err := iamPolicyManagementService.CreatePolicyTemplate(createPolicyTemplateOptions)
+			if err != nil {
+				panic(err)
+			}
+			b, _ := json.MarshalIndent(policyTemplate, "", "  ")
+			examplePolicyS2STemplateID = *policyTemplate.ID
+			examplePolicyS2STemplateVersion = *policyTemplate.Version
+			fmt.Println(string(b))
+
+			// end-create_policy_template
```
Comment (swcolley) [inline iampolicymanagementv1/iam_policy_management_v1_examples_test.go:819]: same as above.

**Review by swcolley — CHANGES_REQUESTED** [review]
Minor change for examples.

File: iampolicymanagementv1/iam_policy_management_v1_examples_test.go (hunk 1 above)
Comment (padamstx) [reply iampolicymanagementv1/iam_policy_management_v1_examples_test.go]: yes, only one code block per operation

File: iampolicymanagementv1/iam_policy_management_v1_examples_test.go [hunk 3]
```diff
@@ -742,6 +753,76 @@ var _ = Describe(`IamPolicyManagementV1 Examples Tests`, func() {
 			Expect(response.StatusCode).To(Equal(201))
 			Expect(policyTemplate).ToNot(BeNil())
 		})
+
+		It(`CreatePolicyS2STemplate request example`, func() {
+			fmt.Println("\nCreatePolicyS2STemplate() result:")
+			// begin-create_authorization_policy_template
```
Comment (padamstx) [inline iampolicymanagementv1/iam_policy_management_v1_examples_test.go:759]:
```suggestion
			// begin-create_policy_template
```
One example code block per operation, and the operation id in the "begin" and "end" comments must match the operationId in the API definition

File: iampolicymanagementv1/iam_policy_management_v1_examples_test.go [hunk 4]
```diff
@@ -820,6 +901,72 @@ var _ = Describe(`IamPolicyManagementV1 Examples Tests`, func() {
 			Expect(policyTemplate).ToNot(BeNil())
 		})
 
+		It(`CreatePolicyS2STemplateVersion request example`, func() {
+			fmt.Println("\nCreatePolicyS2STemplateVersion() result:")
+			// begin-create_authorization_policy_template_version
```
Comment (padamstx) [inline iampolicymanagementv1/iam_policy_management_v1_examples_test.go:906]: Same issue as the create_policy_template examples

**Review by padamstx — CHANGES_REQUESTED** [review]

**Comment by Rajesh-Pirati** [comment]
ran again example tests after fixing the review comments
image

**Pull request merged**